docs/index.md

```diff
@@ -1,98 +1,115 @@
 ---
-title: PMD Introduction
+title: PMD Documentation Index
```

Comment:
Do we need the word "Index" here? I would just name the title then "PMD Documentation". I guess, just "PMD" would be too generic?

Reply:
Hm... I see, we somewhere append always " | PMD Source Code Analyzer" to the title. So, another suggestion: How about just "Documentation Index" - it will become then "Documentation Index | PMD Source Code Analyzer" - not sure.

Reply:
I think the word "Index" belongs here since that's really what that new page is supposed to be. It's not simply because it's the root of the site (an `index.html`), but it really indexes (most) pages of the site and displays them like an index. Well more like a table of contents but whatever. Nevertheless I think you're right that we could rename it to just `Documentation Index`, I had not noticed every page title already has a suffix mentioning PMD!